Client/Ecosia/Extensions/URL+Ecosia.swift

```diff
@@ -23,3 +23,39 @@ extension URL {
         return securityFlags.first(where: { $0 == true }) ?? false
     }
 }
+
+extension URL {
+    
+    /// Retrieves or creates a local URL for a given image set.
+    ///
+    /// This function looks in the app's cache directory to see if an image with the given name and file extension
+    /// already exists. If it does, it returns the URL pointing to this cached image.
+    /// If the image does not exist, it attempts to create it from the app's assets and save it in the cache directory.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the image set.
+    ///   - extension: The file extension of the image (e.g., "png", "jpg", "pdf").
+    ///
+    /// - Returns: The local URL of the image in the cache directory. Returns `nil` if the operation fails.
+    static func localURLForImageset(name: String, withExtension extension: String) -> URL? {
```

Comment:
Why do we need this? Isn't Kingfisher abstracting this already?

Reply:
Didn't know about Kingfisher having this functionality.
I made that so the `WhatsNewItem` will be equipped by a `imageURL: URL?` property.
Having the property as URL will provide us with a scalable model, accepting remote resources as well (in its own DataProvider).
As of now, we only have the `WhatsNewLocalDataProvider` which deals with local items. In order to be able to retrieve the local asset and evaluate the property `imageURL` correctly, I needed this extension.

Reply:
Ok - I think I understand the use-case better now. Though I find the name a bit irritating for what the method does. It's state mutating and the name does not really convey this. As I see it, it serves two purpose. 
1.  Copy an asset catalogue image to the user cache (if not already there)
2. Return the cache URL

Basically you want an URL for a local image, right?

So I want to challenge this. I find it unintuitive to copy a shipped image into the user cache. Wouldn't this cause duplicate data? 

As I understand you want to leverage `URL` to have a scalable model that could either work with local or remote image urls. I see that a local URL is tricky to obtain from the Asset Catalogue. So my suggestion would be to use `data` instead of URL and make it `async`. You can retrieve `data` from `UIImage(named:)` and from a URL request. Does that make sense for your use case?